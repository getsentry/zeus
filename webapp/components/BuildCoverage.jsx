import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router';
import styled from 'styled-components';

import Section from '../components/Section';
import {ResultGrid, Column, Header, Row} from '../components/ResultGrid';
import SyntaxHighlight from '../components/SyntaxHighlight';
import {Tree, Leaf, TreeWrapper, TreeSummary} from '../components/Tree';

const StyledResultGrid = styled(ResultGrid)``;

const StyledHeader = styled(Header)`
  border-color: #ccc;
`;

const StyledRow = styled(Row)`
  border-color: #ccc;
  padding: 0;

  &.good {
    background-color: #e2f5e7;
  }

  &.warn {
    background-color: #ffedde;
  }

  &.bad {
    background-color: #ffe9e9;
  }
`;

const StyledColumn = styled(Column)`
  padding: 8px 4px;
  border-left: 1px solid #ccc;

  &:last-child {
    border-right: 1px solid #ccc;
  }
`;

const FileWithCoverageTable = styled.table`
  width: 100%;

  pre {
    margin-bottom: 0;
    padding: 0: !important;
    background: #fff !important;
    white-space: pre !important;
    overflow-wrap: normal !important;
    word-wrap: normal !important;
  }

  .line-c {
    background-color: #76d392 !important;
  }

  .line-u {
    background-color: #f05b5b !important;
  }

  .line-n {
    background-color: #eee !important;
  }

  td {
    vertical-align: top;
    font-size: 14px !important;
  }

  tr > td:first-child {
    width: 30px;
    text-align: right;
  }

  tr > td:nth-child(2) {
    width: 20px;
  }

  tr > td:nth-child(3) {
    white-space: nowrap;
    overflow: scroll;
  }
`;

class CoveredFile extends Component {
  static propTypes = {
    result: PropTypes.object.isRequired
  };

  render() {
    let {result} = this.props;
    if (!result.file_source) {
      return <p>We were unable to render the source for this file.</p>;
    }
    let lines = result.file_source.split('\n');
    return (
      <FileWithCoverageTable>
        <tr>
          <td>
            {lines.map((l, n) => {
              return <pre key={n}>{n + 1}</pre>;
            })}
          </td>
          <td>
            {lines.map((l, n) => {
              let lineCov = result.coverage.data[n];
              return (
                <pre key={n} className={`line-${(lineCov || 'N').toLowerCase()}`}>
                  &nbsp;
                </pre>
              );
            })}
          </td>
          <td>
            <SyntaxHighlight filename={result.coverage.filename}>
              {result.file_source}
            </SyntaxHighlight>
          </td>
        </tr>
      </FileWithCoverageTable>
    );
  }
}

class CoveredTree extends Component {
  static propTypes = {
    result: PropTypes.object.isRequired,
    location: PropTypes.object.isRequired
  };

  render() {
    let path = this.props.location.pathname;
    let {result} = this.props;
    return (
      <StyledResultGrid>
        <StyledHeader>
          <Column />
          <Column width={250} textAlign="right">
            Lines
          </Column>
        </StyledHeader>
        {result.entries.map(entry => {
          let totalLines = entry.lines_covered + entry.lines_uncovered;
          let pctCovered = parseInt((entry.lines_covered / totalLines) * 100, 10);
          let className;
          if (pctCovered >= 100) {
            className = 'good';
          } else if (pctCovered >= 80) {
            className = 'warn';
          } else {
            className = 'bad';
          }
          return (
            <StyledRow key={entry.path} className={className}>
              <StyledColumn>
                {entry.numTests === 1 ? (
                  <span>{entry.name}</span>
                ) : (
                  <Link to={{pathname: path, query: {parent: entry.path}}}>
                    {entry.name}
                  </Link>
                )}
              </StyledColumn>
              <StyledColumn
                width={120}
                textAlign="right">{`${pctCovered}%`}</StyledColumn>
              <StyledColumn width={120} textAlign="right">
                {`${entry.lines_covered.toLocaleString()} / ${totalLines}`}
              </StyledColumn>
            </StyledRow>
          );
        })}
      </StyledResultGrid>
    );
  }
}

export default class BuildCoverage extends Component {
  static contextTypes = {
    repo: PropTypes.object.isRequired
  };

  static propTypes = {
    location: PropTypes.object.isRequired,
    result: PropTypes.object.isRequired
  };

  render() {
    let {repo} = this.context;
    let {result} = this.props;
    let path = this.props.location.pathname;
    let linesCovered = 0,
      linesTotal = 0;
    result.entries.forEach(e => {
      linesCovered += e.lines_covered;
      linesTotal += e.lines_covered + e.lines_uncovered;
    });
    let pctCovered = parseInt((linesCovered / linesTotal) * 100, 10);

    return (
      <Section>
        <TreeWrapper>
          {!!result.trail.length && (
            <Tree>
              {result.trail.map(crumb => {
                return (
                  <Leaf
                    to={{pathname: path, query: {parent: crumb.path}}}
                    key={crumb.path}>
                    {crumb.name}
                  </Leaf>
                );
              })}
            </Tree>
          )}
          <TreeSummary>
            {!!result.build && (
              <p>
                Data from{' '}
                <Link to={`${repo.full_name}/builds/${result.build.number}`}>{`${
                  repo.owner_name
                }/${repo.name}#${result.build.number}`}</Link>
              </p>
            )}
            <p>{`${pctCovered}% lines covered (${linesCovered} / ${linesTotal})`}</p>
          </TreeSummary>
        </TreeWrapper>
        {result.is_leaf ? (
          <CoveredFile result={result} {...this.props} />
        ) : (
          <CoveredTree result={result} {...this.props} />
        )}
      </Section>
    );
  }
}
