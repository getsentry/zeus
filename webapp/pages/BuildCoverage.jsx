import React from 'react';
import PropTypes from 'prop-types';

import AsyncPage from '../components/AsyncPage';
import CoverageSummary from '../components/CoverageSummary';
import Section from '../components/Section';

export default class BuildCoverage extends AsyncPage {
  static contextTypes = {
    ...AsyncPage.contextTypes,
    build: PropTypes.object.isRequired,
    repo: PropTypes.object.isRequired
  };

  getEndpoints() {
    let {repo} = this.context;
    let {buildNumber} = this.props.params;
    return [['coverage', `/repos/${repo.full_name}/builds/${buildNumber}/file-coverage`]];
  }

  shouldFetchUpdates() {
    return this.context.build.status !== 'finished';
  }

  renderBody() {
    return (
      <Section>
        <CoverageSummary coverage={this.state.coverage} />
      </Section>
    );
  }
}
