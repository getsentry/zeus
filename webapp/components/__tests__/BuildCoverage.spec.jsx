import React from 'react';
import {render} from 'enzyme';

import BuildCoverage from '../BuildCoverage';

const FIXTURE = {
  entries: [
    {
      is_leaf: false,
      lines_covered: 1,
      lines_uncovered: 0,
      diff_lines_covered: 0,
      diff_lines_uncovered: 0,
      name: 'conftest.py',
      path: 'conftest.py'
    },
    {
      is_leaf: false,
      lines_covered: 0,
      lines_uncovered: 8,
      diff_lines_covered: 0,
      diff_lines_uncovered: 8,
      name: 'setup.py',
      path: 'setup.py'
    },
    {
      is_leaf: false,
      lines_covered: 1528,
      lines_uncovered: 2,
      diff_lines_covered: 1528,
      diff_lines_uncovered: 2,
      name: 'tests/zeus',
      path: 'tests/zeus'
    },
    {
      is_leaf: false,
      lines_covered: 125,
      lines_uncovered: 1715,
      diff_lines_covered: 50,
      diff_lines_uncovered: 100,
      name: 'webapp',
      path: 'webapp'
    },
    {
      is_leaf: false,
      lines_covered: 4473,
      lines_uncovered: 1335,
      diff_lines_covered: 50,
      diff_lines_uncovered: 100,
      name: 'zeus',
      path: 'zeus'
    }
  ],
  is_leaf: false,
  trail: []
};

describe('BuildCoverage', () => {
  it('renders with build', () => {
    const tree = render(
      <BuildCoverage
        result={{
          ...FIXTURE,
          build: TestStubs.Build()
        }}
        location={{query: '', pathname: '/path-name'}}
      />,
      {context: {repo: TestStubs.Repository()}}
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders without build', () => {
    const tree = render(
      <BuildCoverage result={FIXTURE} location={{query: '', pathname: '/path-name'}} />,
      {context: {repo: TestStubs.Repository()}}
    );
    expect(tree).toMatchSnapshot();
  });
});
