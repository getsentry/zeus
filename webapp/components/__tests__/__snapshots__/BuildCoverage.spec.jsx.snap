// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildCoverage renders with build 1`] = `
<div
  class="sc-bdVaJa kkDDII"
>
  <div
    class="sc-iwsKbI iGcGwU sc-bdVaJa kkDDII"
  >
    <div
      class="sc-gZMcBi jtErZf"
    >
      <p>
        Data from 
        <a>
          getsentry/zeus#650
        </a>
      </p>
      <p>
        <strong>
          Diff:
        </strong>
         88% lines covered (1628 / 1838)
      </p>
      <p>
        <strong>
          Overall:
        </strong>
         66% lines covered (6127 / 9187)
      </p>
    </div>
  </div>
  <div
    class="sc-gqjmRU hnZXkc sc-ifAKCX kLhjKP"
  >
    <div
      class="sc-VigVT dwUuvj sc-bZQynM dtqgRq ResultGrid-Header sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      />
      <div
        class="sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH fgSvmL"
        style="text-align:right"
        width="250"
      >
        Diff
      </div>
      <div
        class="sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH fgSvmL"
        style="text-align:right"
        width="250"
      >
        Overall
      </div>
    </div>
    <div
      class="good sc-jTzLTM ctnVPw sc-gzVnrw hEczdR ResultGrid-Row sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      >
        <a>
          conftest.py
        </a>
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
         
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        0 / 0
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        100%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        1 / 1
      </div>
    </div>
    <div
      class="bad sc-jTzLTM ctnVPw sc-gzVnrw hEczdR ResultGrid-Row sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      >
        <a>
          setup.py
        </a>
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        0%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        0 / 8
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        0%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        0 / 8
      </div>
    </div>
    <div
      class="warn sc-jTzLTM ctnVPw sc-gzVnrw hEczdR ResultGrid-Row sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      >
        <a>
          tests/zeus
        </a>
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        99%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        1,528 / 1530
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        99%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        1,528 / 1530
      </div>
    </div>
    <div
      class="bad sc-jTzLTM ctnVPw sc-gzVnrw hEczdR ResultGrid-Row sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      >
        <a>
          webapp
        </a>
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        33%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        50 / 150
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        6%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        125 / 1840
      </div>
    </div>
    <div
      class="bad sc-jTzLTM ctnVPw sc-gzVnrw hEczdR ResultGrid-Row sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      >
        <a>
          zeus
        </a>
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        33%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        50 / 150
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        77%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        4,473 / 5808
      </div>
    </div>
  </div>
</div>
`;

exports[`BuildCoverage renders without build 1`] = `
<div
  class="sc-bdVaJa kkDDII"
>
  <div
    class="sc-iwsKbI iGcGwU sc-bdVaJa kkDDII"
  >
    <div
      class="sc-gZMcBi jtErZf"
    >
      <p>
        <strong>
          Diff:
        </strong>
         88% lines covered (1628 / 1838)
      </p>
      <p>
        <strong>
          Overall:
        </strong>
         66% lines covered (6127 / 9187)
      </p>
    </div>
  </div>
  <div
    class="sc-gqjmRU hnZXkc sc-ifAKCX kLhjKP"
  >
    <div
      class="sc-VigVT dwUuvj sc-bZQynM dtqgRq ResultGrid-Header sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      />
      <div
        class="sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH fgSvmL"
        style="text-align:right"
        width="250"
      >
        Diff
      </div>
      <div
        class="sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH fgSvmL"
        style="text-align:right"
        width="250"
      >
        Overall
      </div>
    </div>
    <div
      class="good sc-jTzLTM ctnVPw sc-gzVnrw hEczdR ResultGrid-Row sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      >
        <a>
          conftest.py
        </a>
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
         
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        0 / 0
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        100%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        1 / 1
      </div>
    </div>
    <div
      class="bad sc-jTzLTM ctnVPw sc-gzVnrw hEczdR ResultGrid-Row sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      >
        <a>
          setup.py
        </a>
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        0%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        0 / 8
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        0%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        0 / 8
      </div>
    </div>
    <div
      class="warn sc-jTzLTM ctnVPw sc-gzVnrw hEczdR ResultGrid-Row sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      >
        <a>
          tests/zeus
        </a>
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        99%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        1,528 / 1530
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        99%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        1,528 / 1530
      </div>
    </div>
    <div
      class="bad sc-jTzLTM ctnVPw sc-gzVnrw hEczdR ResultGrid-Row sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      >
        <a>
          webapp
        </a>
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        33%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        50 / 150
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        6%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        125 / 1840
      </div>
    </div>
    <div
      class="bad sc-jTzLTM ctnVPw sc-gzVnrw hEczdR ResultGrid-Row sc-bxivhb bjDEpv sc-bwzfXH ihgQcP"
    >
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH kfUZzQ"
      >
        <a>
          zeus
        </a>
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        33%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        50 / 150
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        77%
      </div>
      <div
        class="sc-fjdhpX hsZIYx sc-EHOje ipitnQ ResultGrid-Column sc-bwzfXH TnVyV"
        style="text-align:right"
        width="120"
      >
        4,473 / 5808
      </div>
    </div>
  </div>
</div>
`;
